### PR TITLE
mame: 0.215 -> 0.216

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

* Bump version
* Set default paths
* Add module for network configuration

Screenshot_20191207_140453

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by gnidorah** [comment]
Going to add module

**Comment by gnidorah** [comment]
@GrahamcOfBorg build mame

File: nixos/modules/programs/mame.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
```
Comment (Lassulus) [inline nixos/modules/programs/mame.nix]: this seems to be more a service than program specific settings

Comment (gnidorah) [reply nixos/modules/programs/mame.nix]: @Lassulus Moved to services, thanks!

File: pkgs/misc/emulators/mame/default.nix [hunk 2]
```diff
@@ -20,30 +22,40 @@ in mkDerivation {
...
 
+  dontWrapQtApps = true;
+
   buildInputs = [ SDL2 SDL2_ttf alsaLib qtbase libXinerama ];
-  nativeBuildInputs = [ python pkgconfig which ];
+  nativeBuildInputs = [ python pkgconfig which makeWrapper ];
 
-  installPhase = ''
-    dest=$out/opt/mame
+  patches = [ ./emuopts.patch ];
```
Comment (Lassulus) [inline pkgs/misc/emulators/mame/default.nix]: can you add a comment why the patch is necessary? Or maybe a source where you got the patch from

Comment (gnidorah) [reply pkgs/misc/emulators/mame/default.nix]: @Lassulus Done, thanks!

File: nixos/modules/services/misc/mame.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      emuAddr = mkOption {
+        type = types.str;
+        description = "IP address you use inside the virtual machine";
+      };
+      hostAddr = mkOption {
+        type = types.str;
+        description = "IP address of host network interface";
+      };
+    };
+  };
```
Comment (Lassulus) [inline nixos/modules/services/misc/mame.nix]: I'm confused on what this service exactly does. What is this mentioned virtual machine? why does the script chmod 666 the tun device? how does networkling happen afterwards?

Comment (gnidorah) [reply nixos/modules/services/misc/mame.nix]: @Lassulus Its a service for mame package (mame is an emulator of old machines). It creates a virtual network device which mame then uses. Without a service network under mame will not work.

Comment (Lassulus) [reply nixos/modules/services/misc/mame.nix]: Well, I get what the goal is, I'm skeptical of the implementation, for example, I wouldn't know where to get the hostAddr or the emuAddr from if I would want to use this.

Comment (gnidorah) [reply nixos/modules/services/misc/mame.nix]: @Lassulus I've added more detailed descriptions. Implementation of the `taputil.sh` is up to mame team. Module just adds required capabilities to mame binary and moves knobs from the script level to nixos config level.

**Review by Lassulus — APPROVED** [review]
LGTM

**Comment by gnidorah** [comment]
@Lassulus thank you!

**Pull request merged**